METRON-1675: Add Pagination to PCAP

sardell opened the pull request:
## Contributor Comments
This PR is a subtask of the PCAP feature branch that adds pagination functionality to the PCAP search results. Unit tests are in place.

You can test by submitting a PCAP query and paging through the results. 

![pcap-paging](https://user-images.githubusercontent.com/7304869/43132105-04234caa-8f3b-11e8-823f-0df76b075554.gif)

## Testing
This has been tested in full dev. In order to test:

1. Spin up full dev.
1. Put pcap data inside of `/apps/metron/pcap/input` in your VM.
1. Start metron-alerts.
1. Sign in and click on the PCAP tab in the upper left hand corner of the screen.
1. Submit a pcap search which will return a set of results from your loaded pcap data, preferably on that will match at least a few dozen results.
1. Page through the results. The lines should update with the next/previous packets. When you download a file, the results should match the current page shown in the UI.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?


### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

cestella commented:
Due to a bug in apache rat, our license checking component, it's ignoring all license checks on typescript files.  I have submitted a PR to up the rat plugin version and have fixed it in master #1126, let's make sure to add license headers here for new typescript files added in this PR though.

mmiklavc commented:
>  Have you included steps to reproduce the behavior or problem that is being changed or addressed?
> 
>  Have you included steps or a guide to how the change may be verified and tested manually?

Can you submit the test script/instructions for this? It's checked off in the list, but has not provided.

sardell commented:
@cestella I've added license headers to all the pcap files in [this commit](https://github.com/apache/metron/pull/1103/commits/dd7c66d17ada01b1e9281f264cd174b0cf8c6639) in PR #1103.

@mmiklavc I've updated my PR description to include a testing section with steps to reproduce. Please let me know if I can clarify any further.

merrimanr commented on metron-interface/metron-alerts/src/app/pcap/model/pcap-status-response.ts:
```diff
@@ -0,0 +1,21 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export class PcapStatusResponse {
```
Isn't this class also defined in pcap.service.ts?  I would prefer to have it as a separate class like it is here but this is missing some fields.  Can we reconcile them here?

sardell replied:
@merrimanr I removed the duplicate model in pcap.service.ts and updated this model's missing/incorrect fields.

merrimanr commented on metron-interface/metron-alerts/src/app/pcap/pcap-pagination/pcap-pagination.component.html:
```diff
@@ -0,0 +1,26 @@
+	OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+  the specific language governing permissions and limitations under the License.
+  -->
+  <!-- <div *ngIf="pagination.total > 0">
```
Looks like commented code was left in.

justinleet replied:
Can we drop this commented code?

merrimanr commented:
Thanks @sardell this looks good to me.  +1

justinleet commented:
+1, thanks for the contribution!

merrimanr commented:
This has been merged @sardell.  Can you close?

mmiklavc commented:
lgtm, +1 via inspection. Thanks for the updates to this PR.
